volatility3/cli/volshell/generic.py

```diff
@@ -312,6 +314,30 @@ def disassemble(self, offset, count=128, layer_name=None, architecture=None):
                 for i in disasm_types[architecture].disasm(remaining_data, offset):
                     print(f"0x{i.address:x}:\t{i.mnemonic}\t{i.op_str}")
 
+    def _get_type_name_with_pointer(
+        self,
+        member_type: Union[
+            str, interfaces.objects.ObjectInterface, interfaces.objects.Template
+        ],
+        depth: int = 0,
+    ) -> str:
+        """Takes a member_type from and returns the subtype name with a * if the member_type is
+        a pointer otherwise it returns just the normal type name."""
+        pointer_marker = "*" * depth
+        try:
+            if member_type.vol.object_class == objects.Pointer:
```

Comment:
```suggestion
            if isinstance(member_type, objects.Pointer):
```
I think this is all I was thinking of?  I can't easily check if this will work, but this should be it before this can be merged?  If you wanna check it, we can get all your hard work into the tree!

Reply:
Thanks very much. I will test that tomorrow and report back.